File: LMIS/LMIS.Web/Modules/ServerTypes/InventoryManage/DownbookshelfsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DownBookshelfsRow } from "./DownBookshelfsRow";

export interface DownBookshelfsColumns {
    DownBookshelfId: Column<DownBookshelfsRow>;
    BookName: Column<DownBookshelfsRow>;
    BookshelfLocation: Column<DownBookshelfsRow>;
    OperateUserName: Column<DownBookshelfsRow>;
    CreateTime: Column<DownBookshelfsRow>;
    UpdateTime: Column<DownBookshelfsRow>;
}

export class DownBookshelfsColumns extends ColumnsBase<DownBookshelfsRow> {
    static readonly columnsKey = 'InventoryManage.DownBookshelfs';
    static readonly Fields = fieldsProxy<DownBookshelfsColumns>();
}

File: LMIS/LMIS.Web/Modules/InventoryManage/DownBookshelfs/DownBookshelfsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.InventoryManage.Columns;

[ColumnsScript("InventoryManage.DownBookshelfs")]
[BasedOnRow(typeof(DownBookshelfsRow), CheckNames = true)]
public class DownBookshelfsColumns
{
    [EditLink,  AlignRight]
    public long DownBookshelfId { get; set; }
    public string BookName { get; set; }
    public string BookshelfLocation { get; set; }
    public string OperateUserName { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime CreateTime { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}